lms/djangoapps/mobile_api/mobile_platform.py

```diff
@@ -0,0 +1,59 @@
+"""
+Platform related operations for Mobile APP
+"""
+import abc
+import re
+
+
+class MobilePlatform:
+    """
+    MobilePlatform class creates an instance of platform based on user agent and supports platform
+    related operations
+    """
+    __metaclass__ = abc.ABCMeta
+    version = None
+
+    def __init__(self, version):
+        self.version = version
+
+    @classmethod
+    def create_instance(cls, user_agent, user_agent_regex):
+        """ Returns platform instance if user_agent matches with USER_AGENT_REGEX """
+        match = re.search(user_agent_regex, user_agent)
+        if match:
+            return cls(match.group('version'))
+
+    @classmethod
+    def get_instance(cls, user_agent):
+        """
+        It creates an instance of one of the supported mobile platforms (i.e. iOS, Android) by regex comparison
+        of user-agent.
+
+        Parameters:
+            user_agent: user_agent of mobile app
+
+        Returns:
+            instance of one of the supported mobile platforms (i.e. iOS, Android)
+        """
+        for subclass in PLATFORM_CLASSES.values():
+            instance = subclass.create_instance(user_agent, subclass.USER_AGENT_REGEX)
+            if instance:
+                return instance
+
+
+class Ios(MobilePlatform):
```

Comment:
Nit: The capitalization for `iOS` here is hard to read since it's not the iOS standard.  Rather than following python's class naming standard here, I'd suggest silencing pylint and calling it `IOS`.